array.c

```diff
@@ -1567,15 +1606,16 @@ ary_ensure_room_for_unshift(VALUE ary, int argc)
 
 /*
  *  call-seq:
- *     ary.unshift(obj, ...)  -> ary
- *     ary.prepend(obj, ...)  -> ary
+ *    array.unshift(*objects) -> self
+ *    array.prepend(*objects) -> self
  *
- *  Prepends objects to the front of +self+, moving other elements upwards.
- *  See also Array#shift for the opposite effect.
+ *  #prepend is an alias for \#unshift.
  *
- *     a = [ "b", "c", "d" ]
- *     a.unshift("a")   #=> ["a", "b", "c", "d"]
- *     a.unshift(1, 2)  #=> [ 1, 2, "a", "b", "c", "d"]
+ *  Prepends the given +objects+ to +self+:
+ *    a = [:foo, 'bar', 2]
+ *    a1 = a.unshift(:bam, :bat)
```

Comment:
This assignment + `unshift` path add more confusion than help. Keeping the original would be better, IMO.
If we need to point it returns a modified self, we may be better adding them as a phrase.

Reply:
I have 5 other such as this here in Array, and 20+ over at Hash, all reviewed by multiple reviewers such as @drbrain and @ioquatix.  I am not confortable either to become inconsistent or to change all the others.  Opinion?

Reply:
it was just an opinion, if was already approved by other ruby members, please disregard.